src/client/InteractionClient.js

```diff
@@ -0,0 +1,232 @@
+ *   } else {
+ *     interaction.reply('hi!');
+ *   }
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
+   * @param {undefined} client For internal use.
+   */
+  constructor(options, client) {
+    super(options);
+
+    Object.defineProperty(this, 'token', {
+      value: options.token,
+      writable: true,
+    });
+
+    if (client) {
+      this.client = client;
+    } else {
+      this.client = this;
+      this.interactionClient = this;
+
+      Object.defineProperty(this, 'applicationID', {
+        value: options.clientID,
+        writable: true,
+      });
+
+      Object.defineProperty(this, 'publicKey', {
+        value: options.publicKey ? Buffer.from(options.publicKey, 'hex') : undefined,
+        writable: true,
+      });
+    }
+  }
+
+  /**
+   * Fetch registered slash commands.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]}
+   */
+  async fetchCommands(guildID) {
+    let path = this.client.api.applications(this.applicationID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const commands = await path.commands.get();
+    return commands.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Set all the commands for the application or guild.
+   * @param {Object[]} commands The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]} The commands.
+   */
+  async setCommands(commands, guildID) {
+    let path = this.client.api.applications(this.applicationID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const cs = await path.commands.post({
+      data: commands.map(transformCommand),
+    });
+    return cs.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Create a command.
+   * @param {Object} command The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand} The created command.
+   */
+  async createCommand(command, guildID) {
+    let path = this.client.api.applications(this.clientID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const c = await path.commands.post({
+      data: transformCommand(command),
+    });
+    return new ApplicationCommand(this, c, guildID);
+  }
+
+  handle(data) {
+    switch (data.type) {
+      case InteractionType.PING:
+        return {
+          type: InteractionResponseType.PONG,
+        };
+      case InteractionType.APPLICATION_COMMAND: {
+        let timedOut = false;
+        let resolve;
+        const directPromise = new Promise(r => {
+          resolve = r;
+          this.client.setTimeout(() => {
+            timedOut = true;
+            r({
+              type: InteractionResponseType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+            });
+          }, 250);
+        });
+
+        const syncHandle = {
+          acknowledge() {
```

Comment:
This should take options (and is already passed from CommandInteraction#acknowledge) to allow deferred ephemeral responses as it cannot be changed in the edit.